MoviesExample/app/src/main/java/com/novoda/demo/movies/MoviesViewModel.java

```diff
@@ -0,0 +1,51 @@
+package com.novoda.demo.movies;
+
+import android.arch.core.util.Function;
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.Transformations;
+import android.arch.lifecycle.ViewModel;
+
+import com.novoda.demo.movies.model.Movie;
+import com.novoda.demo.movies.model.Video;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class MoviesViewModel extends ViewModel {
+
+    private final MovieService movieService;
+    private MoviesSate moviesSate = new MoviesSate(new ArrayList<Movie>(), 1);
+
+    MoviesViewModel(MovieService movieService) {
+        this.movieService = movieService;
+    }
+
+    public LiveData<MoviesSate> moviesLiveData() {
+        return Transformations.map(movieService.loadMore(moviesSate.pageNumber()),
```

Comment:
`Transformations` is another utility class that comes with the Lifecycle architecture components library. 
From the [documentation](https://developer.android.com/reference/android/arch/lifecycle/Transformations):
> You can use transformation methods to carry information across the observer's lifecycle. The transformations aren't calculated unless an observer is observing the returned LiveData object.
> Because the transformations are calculated lazily, lifecycle-related behavior is implicitly passed down without requiring additional explicit calls or dependencies.

Here we use `Transformations.map` to transform the _raw_ API data into a domain data model. This is a very simple use case but it serves the purpose. 

One interesting thing about `Transformations.map` is that it implicitly passes down the lifecycle owner. The LiveData object returned by `movieService.loadMore` will be linked to the same lifecycle owner that calls `moviesLiveData()`

Reply:
So the benefit of the transformation API is to give it a chance to ... not do the transformation if there's nothing observing anymore?

I thought that onchanged would not be called if nothing was observing the livedata.

Reply:
Good question.
`Transformations.map` is just a map, the benefit of using it is to express intention, keep things cleaner and share a LifecycleOwner between observed liveDatas

For example, you could achieve the same result by observing the `movieService` LiveDataA from the viewModel and using the observed result to update a LiveDataB observed from `MainActivity`
The problem with this second approach is the lifecycleOwner you need to pass to observe LiveDataA. `MainActivity` is a lifecycleOwner so it can observe LiveDataB from the ViewModel with no problem. But `MoviesViewModel` is *not* a lifecycleOwner, so in order to observe LiveDataA from `movieService` you'll need to pass a reference of `MainActivity` to `MoviesViewModel` so it can use it to observe LiveDataB.

By using `Transformations.map` the first lifecycleOwner is shared among all chained LiveDatas

Reply:
ah right, so in this case, it's just demonstrating how to use the API, but it's not specifically beneficial in our setup here?

Reply:
The mapping it's doing is trivial, so in that sense it's not adding much benefit but rather demonstrating how to use the API
But it also allows us to avoid passing a lifecycleOwner to the viewModel just to do a second observation. And this is a real benefit

Reply:
ohhh seeen. because we're mapping from `LiveData<X>` -> `LiveData<Y>`